server/src/test/java/org/elasticsearch/health/HealthServiceTests.java

```diff
@@ -114,11 +121,177 @@ public void testMissingComponentOrIndicator() {
+        }
+        {
+            HealthComponentResult component2 = health.stream().filter(result -> result.name().equals("component2")).findAny().orElseThrow();
+            assertThat(component2.status(), is(equalTo(GREEN)));
+            assertThat(component2.indicators(), is(notNullValue()));
+            assertThat(component2.indicators(), contains(indicator3));
+        }
+
+        // Getting single component returns preflight result mixed in with appropriate component
+        health = service.getHealth("component1", null, false);
+        assertThat(health.size(), is(equalTo(1)));
+        {
+            HealthComponentResult component1 = health.stream().filter(result -> result.name().equals("component1")).findAny().orElseThrow();
+            assertThat(component1.status(), is(equalTo(YELLOW)));
+            assertThat(component1.indicators(), is(notNullValue()));
+            assertThat(component1.indicators(), containsInAnyOrder(preflight1, indicator1, indicator2));
+        }
+
+        // Getting single indicator returns correct indicator still
+        health = service.getHealth("component1", "indicator2", false);
+        assertThat(health.size(), is(equalTo(1)));
+        {
+            HealthComponentResult component1 = health.stream().filter(result -> result.name().equals("component1")).findAny().orElseThrow();
+            assertThat(component1.indicators(), is(notNullValue()));
+            assertThat(component1.indicators(), contains(indicator2));
+        }
+
+        // Getting single preflight indicator returns preflight indicator correctly
+        health = service.getHealth("component1", "preflight1", false);
+        assertThat(health.size(), is(equalTo(1)));
+        {
+            HealthComponentResult component1 = health.stream().filter(result -> result.name().equals("component1")).findAny().orElseThrow();
+            assertThat(component1.indicators(), is(notNullValue()));
+            assertThat(component1.indicators(), contains(preflight1));
+        }
+    }
+
+    private void assertIndicatorIsUnknownStatus(HealthIndicatorResult result) {
+        assertThat(result.status(), is(equalTo(UNKNOWN)));
+        assertThat(result.summary(), is(equalTo(HealthService.UNKNOWN_RESULT_SUMMARY)));
+    }
+
+    public void testPreflightIndicatorFailureTriggersUnknownResults() {
+        var preflight1 = new HealthIndicatorResult("preflight1", "component1", RED, null, null, null, null);
+        var preflight2 = new HealthIndicatorResult("preflight2", "component2", GREEN, null, null, null, null);
+        var indicator1 = new HealthIndicatorResult("indicator1", "component1", GREEN, null, null, null, null);
+        var indicator2 = new HealthIndicatorResult("indicator2", "component1", YELLOW, null, null, null, null);
+        var indicator3 = new HealthIndicatorResult("indicator3", "component2", GREEN, null, null, null, null);
```

Comment:
Could we generally use real values in tests? 

Seeing a failure that has component2 and indicator3 in the error message is rather hard to "parse"

Reply:
How real should these be? I want to make sure that preflight indicators that are in the same component get correctly mixed together in the results, but the master indicator doesn't share a component with anything right now. For now I'll give them some hypothetical names, but if you'd like them to be more real than that we can discuss further.